src/openfermion/measurements/_probability_distributions.py

```diff
@@ -0,0 +1,379 @@
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+"""
+Classes to store probability distributions for estimation
+of random variables.
+"""
+import numpy
+from scipy import sparse
+from numpy import cos, sin, pi
+from openfermion.config import EQ_TOLERANCE
+
+
+class FourierProbabilityDist(object):
+    """
+    Stores an approximate probability distribution of our knowledge of
+    a quantum state in the eigenbasis of a unitary operator.
+
+    A generic quantum state may be written in the eigenbasis of a unitary
+    operator U,
+
+    $|\psi\rangle = sum_j a_j |phi_j\rangle$,
+
+    where psi is the quantum state, |phi_j\rangle are the eigenstates of U,
+    labeled by their (unknown) eigenvalues
+
+    $U|phi_j\rangle = phi_j|phi_j\rangle$,
+
+    and a_j are the (unknown) corresponding amplitudes.
+
+    This class stores a distribution P(phi_0, phi_1,... , A_0, A_1,...)
+    (where A_j = |a_j|^2) of our knowledge of these variables.
+
+    In particular, it approximates this distribution as:
+    P(phi_0, phi_1,... , A_0, A_1,...)=P(A_0, A_1,...)*\prod_jP(\phi_j).
```

Comment:
Is P(A_0, A_1, ...) a normal distribution? If so then document it here.

Reply:
See below. It should be, and i'll fix it up later.